### DISCUSSION TITLE
How should we handle ingested fields that are identical but have different field names?

### DISCUSSION THREAD

**Discussion by razakpm** [diabetes-research/help-desk-non-phi#6] [Q&A] [answered]
How should we manage ingested fields that are identical in content but have different field names than those specified in the file format on the Getting Started page of the DRH Edge tool? Looking for guidance on mapping or handling these fields efficiently.

**Comment by MeetAnithaVarghese** [comment, marked as answer]
# Using Surveilr for Data Ingestion and Transformation

We use a tool named **Surveilr** for the ingestion and transformation of files into a database format. The tool automatically converts each file, using the file headers as column names in the resulting database.

## Example

- If your file is named `patient` and contains a header `patient_id`, Surveilr will automatically create a database table with the name `uniform_resource_<filename>` and `patient_id` as the column name.

## How to Modify the Format
If you require a different column name (e.g., `patient_id` instead of `participant_id`), or if the filename used for participant data is different (e.g., `patient.csv`), you can make adjustments **after the ingestion process**.

## Steps

### 1. Ingestion & Transformation
Once Surveilr performs the ingestion, it transforms the file data into a database format based on the provided file headers.

#### Sample steps for ingestion and local UI setup:

> **Note:** The `study-files` directory contains your files.

```bash
surveilr ingest files -r study-files/ && surveilr orchestrate transform-csv
```
### Deidentification and Verification & Validation
Separate SQL files containing SQL for de-identification and Verification & Validation are available. These files are executed through Surveilr.

### SQL for DB views(Data display in local UI)

An SQL file prepares necessary database views for data display in the local UI. Example: `stateless.SQL`. Based on your structural changes, the database view needs to be modified as needed. A sample is shown below.

For our local UI setup, we use an automation script. The `stateless.SQL` is invoked through the `package.sql.ts`. The following changes should be made in that SQL.

You can modify the  SQL file to modify the view for content display. For example, if you want to rename `participant_id` to `patient_id`, you can perform the necessary SQL operation.
The filename and column names can be changed in the SQL view, which will help render the data in the UI.

#### Sample SQL in 
[different-patient-column-names.sql](https://raw.githubusercontent.com/surveilr/www.surveilr.com/main/lib/service/diabetes-research-hub/cookbook/different-patient-column-names.sql)
```sql
-- Drop and recreate the participant_data view
DROP VIEW IF EXISTS drh_participant_data;
CREATE VIEW drh_participant_data AS
SELECT
  patient_id AS participant_id, study_id, site_id, diagnosis_icd, med_rxnorm,
  treatment_modality, gender, race_ethnicity, age, bmi, baseline_hba1c,
  diabetes_type, study_arm
FROM uniform_resource_patient;
```

### 3. View Data in Local UI

Automation script performs all the above activities
```bash
$ deno run -A https://raw.githubusercontent.com/surveilr/www.surveilr.com/main/lib/service/diabetes-research-hub/drhctl.ts <Foldername>
```

### R&D on Diverse Datasets

We are conducting research and development (R&D) on various datasets to explore and test the necessary changes for data conversion, transformation, cleaning, and de-identification processes. This is currently in the research phase .
